dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/tool/DeploymentProblemTest.java

```diff
@@ -0,0 +1,80 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.mcp.internal.fat.tool;
+
+import static com.ibm.websphere.simplicity.ShrinkHelper.DeployOptions.DISABLE_VALIDATION;
+import static com.ibm.websphere.simplicity.ShrinkHelper.DeployOptions.SERVER_ONLY;
+import static org.junit.Assert.assertTrue;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+import io.openliberty.mcp.internal.fat.tool.duplicateToolErrorTestApps.DuplicateToolErrorTest;
+import io.openliberty.mcp.internal.fat.tool.duplicateToolErrorTestApps.DuplicateToolErrorTest2;
+import io.openliberty.mcp.internal.fat.utils.HttpTestUtils;
+
+/**
+ *
+ */
+@RunWith(FATRunner.class)
+public class DeploymentProblemTest extends FATServletClient {
+
+    @Server("mcp-server")
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, "DeploymentProblemTest.war").addPackages(true, DuplicateToolErrorTest.class.getPackage(),
+                                                                                                      DuplicateToolErrorTest2.class.getPackage());
+        ShrinkHelper.exportDropinAppToServer(server, war, DISABLE_VALIDATION, SERVER_ONLY);
+        server.startServer();
+    }
+
+    @AfterClass
+    public static void teardown() throws Exception {
+        server.stopServer("CWWKZ0002E");
+    }
+
+    @Test
+    public void testDuplicateToolDeploymentError() throws Exception {
+        String request = """
+                        {
+                          "jsonrpc": "2.0",
+                          "id": 1,
+                          "method": "tools/list",
+                          "params": {
+                            "cursor": "optional-cursor-value"
+                          }
+                        }
+                        """;
+
+        boolean deploymentErrorOccured = false;
+
+        try {
+            HttpTestUtils.callMCP(server, "/toolTest", request);
+        } catch (Exception e) {
+            if (!server.findStringsInLogs("More than one MCP tool has the same name").isEmpty()) {
+                deploymentErrorOccured = true;
+            }
+        }
+
+        assertTrue("Expected a deployment error due to duplicate MCP Tools being present, no error was present", deploymentErrorOccured);
```

Comment:
It's not necessary to make a request here, the error should be output when the application starts up. You can just check the logs.

However, we should check for all of the expected log lines.

Reply:
This hasn't been addressed, but let's get this merged and fix this in a follow up PR.